hooks/use-hypercert-client.ts

```diff
@@ -4,15 +4,21 @@ import { useEffect, useState } from "react";
 import { useAccount, useWalletClient } from "wagmi";
 import { ENVIRONMENT, SUPPORTED_CHAINS } from "@/configs/constants";
 import { EvmClientFactory } from "@/lib/evmClient";
+import { PublicClient } from "viem";
 
 export const useHypercertClient = () => {
   const { data: walletClient } = useWalletClient();
   const { isConnected } = useAccount();
   const [client, setClient] = useState<HypercertClient>();
 
-  const publicClient = walletClient?.chain.id
-    ? EvmClientFactory.createClient(walletClient.chain.id)
-    : undefined;
+  let publicClient: PublicClient | undefined;
```

Comment:
Personally I would try to have as little uninitialized variables in code as possible and just move this out to a function that returns `PublicClient | undefined`.